hypershift-operator/controllers/hostedcluster/internal/platform/openstack/openstack.go

```diff
@@ -39,31 +38,76 @@ func New(capiProviderImage string) *OpenStack {
+		openStackCluster.Spec.IdentityRef = capo.OpenStackIdentityReference{
+			Name:      openStackPlatform.CloudsYamlSecret.Name,
 			CloudName: cloudName,
 		}
 	}
 
-	if _, err := createOrUpdate(ctx, client, openstackCluster, func() error {
-		openstackCluster.Annotations = map[string]string{
-			capiv1.ManagedByAnnotation: "external",
-		}
-		// TODO(maysa): figure out why this is not being set on the object
-		openstackCluster.Status.Ready = true
+	if _, err := createOrUpdate(ctx, client, openStackCluster, func() error {
+		reconcileOpenStackCluster(hcluster, openStackCluster, openStackPlatform)
 		return nil
 	}); err != nil {
 		return nil, err
 	}
-	return openstackCluster, nil
+	return openStackCluster, nil
+}
+
+func reconcileOpenStackCluster(hcluster *hyperv1.HostedCluster, openStackCluster *capo.OpenStackCluster, openStackPlatform *hyperv1.OpenStackPlatformSpec) {
+
+	if len(openStackPlatform.Subnets) > 0 {
+		openStackCluster.Spec.Subnets = make([]capo.SubnetParam, len(openStackPlatform.Subnets))
+		for i := range openStackPlatform.Subnets {
+			openStackCluster.Spec.Subnets[i] = capo.SubnetParam{
+				ID: openStackPlatform.Subnets[i].ID,
+			}
+		}
+	} else {
+		machineNetworks := hcluster.Spec.Networking.MachineNetwork
+		openStackCluster.Spec.ManagedSubnets = make([]capo.SubnetSpec, len(machineNetworks))
+		// Only one Subnet is supported in CAPO
+		openStackCluster.Spec.ManagedSubnets[0] = capo.SubnetSpec{
+			CIDR: hcluster.Spec.Networking.MachineNetwork[0].CIDR.String(),
+		}
+		for i := range openStackPlatform.ManagedSubnets {
+			openStackCluster.Spec.ManagedSubnets[i].DNSNameservers = make([]string, len(openStackPlatform.ManagedSubnets[i].DNSNameservers))
+			openStackCluster.Spec.ManagedSubnets[i].DNSNameservers = openStackPlatform.ManagedSubnets[i].DNSNameservers
+			allocationPools := openStackPlatform.ManagedSubnets[i].AllocationPools
+			openStackCluster.Spec.ManagedSubnets[i].AllocationPools = make([]capo.AllocationPool, len(allocationPools))
+			for j := range allocationPools {
+				openStackCluster.Spec.ManagedSubnets[i].AllocationPools[j] = capo.AllocationPool{
+					Start: allocationPools[j].Start,
+					End:   allocationPools[j].End,
+				}
+			}
+		}
+	}
+	if openStackPlatform.Router != nil {
+		openStackCluster.Spec.Router = &capo.RouterParam{ID: openStackPlatform.Router.ID}
+	}
+	if openStackPlatform.Network != nil {
+		openStackCluster.Spec.Network = &capo.NetworkParam{ID: openStackPlatform.Network.ID}
+	}
+
+	if openStackPlatform.NetworkMTU != nil {
+		openStackCluster.Spec.NetworkMTU = openStackPlatform.NetworkMTU
+	}
+	if openStackPlatform.ExternalNetwork != nil {
+		openStackCluster.Spec.ExternalNetwork = &capo.NetworkParam{ID: openStackPlatform.ExternalNetwork.ID}
+	}
+	if openStackPlatform.DisableExternalNetwork != nil {
+		openStackCluster.Spec.DisableExternalNetwork = openStackPlatform.DisableExternalNetwork
+	}
+	openStackCluster.Spec.ManagedSecurityGroups = &capo.ManagedSecurityGroups{}
```

Comment:
I didn't add additional SG rules as I'm not sure which additional ones might be required.
I thought about adding once we cover nodePools. Thoughts?
@EmilienM

Reply:
we don't need SGs until we get workers deployed, so yeah we can wait.
Now, regarding the rules themselves, we'll want to compare what is managed by CAPO: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/pkg/cloud/services/networking/securitygroups_rules.go and what we need in OpenShift.

Reply:
This is good for now.